Add scripts for Mac and Linux

guicaulada opened the pull request:
This pull request includes several new scripts, equivalents of the existing batch scrits but for shell. Due to the number of scripts after this change I reorganized the files in the Tools folder and updated existing scripts (mainly LuaConverter.bat) to use the new folder structure.

Every script work the same as before, you execute them from inside the tools folder but add the `./script/` before the file name.

llchrisll commented:
Thanks for the contributation, now if someone with Linux or Mac can test this, so I can merge it would be nice.

llchrisll commented:
If you wouldn't mind, updating the Generator's to the latest?
I changed a bit, since you made them.

guicaulada commented:
@llchrisll Done!

llchrisll commented:
That's true, but since I didn't have any knowledge in that and I wanted it to be executed without any additional programs (python as example), I went with .bat.
My question can .sh also be executed on windows or only linux/mac?
I didn't try them yet tbh.

llchrisll commented:
After testing them a bit, I realized you didn't change the path in regards to the new scripts folder, well you kinda mentioned it.
But as I like to keep the actual scripts in the main folder of "Tools", I moved them over locally.
![image](https://github.com/user-attachments/assets/153b0af2-0aa0-4a41-a964-cd95dd201513)
Also I added a simple echo after the Client folder was generated, as confirmation for the user :).
![image](https://github.com/user-attachments/assets/b490fbae-d1be-424a-9282-2cc40eea9670)

llchrisll commented on Tools/scripts/AdditionsGenerator.sh:
```diff
@@ -0,0 +1,618 @@
+                    16) cp -f "$sourceDataPath/resnametable.txt" "$destinationDataPath/" && datac[16]="[ Copied ]" ;;
+                    17) cp -f "$sourceDataPath/viewpointtable.txt" "$destinationDataPath/" && datac[17]="[ Copied ]" ;;
+                esac
+            done
```
Add a `read -p "Press enter to continue"` after each process, in case there are errors.

llchrisll replied:
Note: I'm just learning a bit about this review thingy, never really used it.
So don't take these too seriously, I will write comments first about it xD.

guicaulada replied:
That's fine! Thank you for your review, I will address the comments as soon as possible. :)

llchrisll commented:
> > My question can .sh also be executed on windows or only linux/mac?
> 
> Shell script could be executed on Windows nowadays, but that will require WSL to be setup.
> 
> Most machines, including Windows, comes with Python now, so that's why I suggested it as an option, but if you have no knowledge on that it might be hard to support it as the core maintainer of the project, and it would require Python installation on machines where for some reason it isn't installed.
> 
> Not sure if this makes it more difficult but PowerShell is another option which requires zero dependencies and is cross-platform and can be executed in both Linux and Mac for simpler implementations like these, the only issue I have with it is that it's super verbose and often looks more complex than it is, but if you like that idea I could update the PR and we give it a try :)

I can execute shell without issues on win11, but had to install Python.

in the end I just need something which is not to hard to update on my own, I mean shell looks similar to bat, so getting used to that will not be hard imo.

guicaulada commented:
You can execute shell without issues on Win 11? Huh, maybe I'm a bit outdated and Windows just fully integrated it now.

If that's the case would you prefer this PR replaced the .bat scripts with .sh scripts or kept them side by side?

llchrisll commented:
I would remove the bat then, since keeping both versions would be stupid.

guicaulada commented:
Hi @llchrisll,

Following up on the shell script conversion, I've incorporated some improvements to enhance compatibility, user feedback, and maintainability:

1.  **Improved Error Handling & Pausing:**
    *   The core file/directory copy functions (`copy_file`, `copy_dir_recursive` now located in `Helper.sh`) have been updated.
    *   They now explicitly check for:
        *   Source file/directory existence.
        *   Success of destination directory creation (`mkdir -p`).
        *   Success of the `cp` command itself.
    *   If any of these steps fail, a specific error message is printed, and the script pauses using `pause_script`, allowing the user to see the error before proceeding. This applies to `AdditionsGenerator.sh`, `ClientGenerator.sh`, and `CLSGenerator.sh`.

2.  **Accurate Status Reporting:**
    *   In `AdditionsGenerator.sh` and `CLSGenerator.sh`, the menu status indicators (e.g., `omaps[N]`, `cls[N]`) are now updated more accurately.
    *   They will display `[ Failed ]` if the corresponding copy operation failed, instead of always showing `[ Copied ]`.
    *   The "All in One Package" options now also attempt to update the status of the individual items they encompass, marking them as `[ Copied ]` or `[ Failed ]` based on the outcome during the "All in One" run.

3.  **Dynamic Korean Directory Detection (`AdditionsGenerator.sh`):**
    *   To handle potential encoding/normalization issues (like `유저인터페이스` vs. `À¯ÀúÀÎÅÍÆäÀÌ½º`), the script now tries to detect the correct directory name automatically at startup.
    *   The detection order is:
        1.  Environment variable `KOREAN_DIR_NAME` (if set and points to a valid directory).
        2.  Default Korean name (`유저인터페이스`).
        3.  Alternative/Mojibake name (`À¯ÀúÀÎÅÍÆäÀÌ½º`).
    *   If none are found, it defaults to `유저인터페이스` and warns the user, suggesting they set the environment variable if errors occur later.

4.  **Robust Lua Compiler Detection (`LuaConverter.sh`):**
    *   The logic for finding `luac` has been improved.
    *   It distinguishes between a compiler being "not found" and being "found but not executable" (e.g., due to permissions or incorrect architecture like a Windows `.exe` on Linux/macOS).
    *   It attempts a silent test execution (`luac -v`) on local `./luac` and `./luac.exe` before using them.
    *   The search order remains: `LUAC_PATH` env var -> working `./luac` -> working `./luac.exe` -> system `PATH`.

5.  **Refactoring into `Helper.sh`:**
    *   Common functions (`pause_script`, `copy_file`, `copy_dir_recursive`) have been moved from the individual generator scripts into `Tools/Helper.sh`.
    *   `AdditionsGenerator.sh`, `ClientGenerator.sh`, `CLSGenerator.sh`, and `LuaConverter.sh` now `source` this helper file, reducing code duplication and improving maintainability.

These changes should significantly improve the reliability and user experience of the shell scripts. Please let me know if you have any feedback or if any of these scripts fail in your Windows environment.

I don't have a Windows workstation anymore, so testing this locally is difficult.

llchrisll commented:
Okay for some reason it wants to open .sh with "Git for Windows" now lol.
I think after merging I will need to fix some issues I realized just now.
Since I cleaned up my CLS from unneeded files like _f.lub files, I might have forgotten to change CLSGenerator.bat xD.
Will check this now.
But other than that I don't see any issues in exection of saied shell scripts.
Good Job, now I only need to understand the part where I can make adjustments myself, like new client folders and stuff :P.

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+                item_failed=false
+                copy_file "$sourceDataPath/cls/accessoryid.lub" "$destinationDataPath/cls/accessoryid.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/accname.lub" "$destinationDataPath/cls/accname.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/accname_f.lub" "$destinationDataPath/cls/accname_f.lub" || item_failed=true
```
Remove this.

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL65

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+                copy_file "$sourceDataPath/cls/accessoryid.lub" "$destinationDataPath/cls/accessoryid.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/accname.lub" "$destinationDataPath/cls/accname.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/accname_f.lub" "$destinationDataPath/cls/accname_f.lub" || item_failed=true
+                copy_file "$sourceDataPath/datainfo/TB_Layer_Priority.lub" "$destinationDataPath/datainfo/TB_Layer_Priority.lub" || item_failed=true
```
Replace with 
```
copy_file "$sourceDataPath/cls/tb_layer_priority.lub" "$destinationDataPath/cls/tb_layer_priority.lub" || item_failed=true
```

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bR65

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+                ;;
+            4) # HatEffects
+                item_failed=false
+                copy_file "$sourceDataPath/hateffectinfo/hateffectinfo.lub" "$destinationDataPath/hateffectinfo/hateffectinfo.lub" || item_failed=true
```
Remove this.

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL80

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+            5) # NPC/Mob and Pets
+                item_failed=false
+                copy_file "$sourceDataPath/cls/jobname.lub" "$destinationDataPath/cls/jobname.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/jobname_f.lub" "$destinationDataPath/cls/jobname_f.lub" || item_failed=true
```
Remove this.

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL88

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+                ;;
+            6) # Lapine Boxes
+                item_failed=false
+                copy_file "$sourceDataPath/datainfo/lapineddukddakbox.lub" "$destinationDataPath/datainfo/lapineddukddakbox.lub" || item_failed=true
```
Remove lines 96 and 97

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL96-L97

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+            7) # Signboards
+                item_failed=false
+                copy_file "$sourceDataPath/cls/signboardlist.lub" "$destinationDataPath/cls/signboardlist.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/signboardlist_f.lub" "$destinationDataPath/cls/signboardlist_f.lub" || item_failed=true
```
Remove this.

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL106

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+                item_failed=false
+                copy_file "$sourceDataPath/cls/spriterobeid.lub" "$destinationDataPath/cls/spriterobeid.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/spriterobename.lub" "$destinationDataPath/cls/spriterobename.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/spriterobename_f.lub" "$destinationDataPath/cls/spriterobename_f.lub" || item_failed=true
```
Remove this.

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL114

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+                copy_file "$sourceDataPath/cls/spriterobename.lub" "$destinationDataPath/cls/spriterobename.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/spriterobename_f.lub" "$destinationDataPath/cls/spriterobename_f.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/transparentitem.lub" "$destinationDataPath/cls/transparentitem.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/transparentitem_f.lub" "$destinationDataPath/cls/transparentitem_f.lub" || item_failed=true
```
Remove this.

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL116

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+            9) # Weapons
+                item_failed=false
+                copy_file "$sourceDataPath/cls/weapontable.lub" "$destinationDataPath/cls/weapontable.lub" || item_failed=true
+                copy_file "$sourceDataPath/cls/weapontable_f.lub" "$destinationDataPath/cls/weapontable_f.lub" || item_failed=true
```
Remove this.

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL123

llchrisll commented on Tools/CLSGenerator.sh:
```diff
@@ -0,0 +1,180 @@
+                ;;
+            10) # World Map
+                item_failed=false
+                copy_file "$sourceDataPath/cls/worldviewdata_f.lub" "$destinationDataPath/cls/worldviewdata_f.lub" || item_failed=true
```
Remove this.

guicaulada replied:
Done on https://github.com/llchrisll/ROenglishRE/pull/100/commits/0e67ea0ed5152e0ed3775913935022d1643dbb67#diff-51a4c473a095a632f039964dc2a80c0e4c5794873ab1cfccfcbdffdf7114bb3bL129

llchrisll commented:
Also I feel like that the Helper.sh should be placed in a different folder, so users don't open it out of curiosity or like that.

guicaulada commented:
Thank you for the review! I have updated the CLSGenerator.sh file and linked to the each change on the comments above. 

It's weird that Windows tried to use Git for Windows, but I think you could use `bash <script name>` to execute? If that's the case we could update the documentation to reflect that :)

Let me know if there's anything else we can improve or fix! 👍

eleriaqueen commented:
Hi, something minor: 
There's still a mention of a ".bat" file, namely "ClientGenerator.bat", left on line 833 of "AdditionsGenerator".
![Screenshot_20250527_170034](https://github.com/user-attachments/assets/a39da757-d1cf-4672-ab61-59cae27fc96e)

guicaulada commented:
Oh right! I forgot to update the note, I will do that right away!

llchrisll commented:
Now after some research and feedback on my discord server, we found the following:
`Unfortunately, there's no built-in way to run shell scripts on a Windows machine. You need to use third-party tools like Git Bash, WSL, Cygwin, or MinGW. `

Meaning we need to keep that bat files for users which don’t have said programs installed.

guicaulada commented:
We could re-add the .bat files back, but it would reintroduce that issue with mantaining all scripts in sync.

I'm wondering, what do you think about PowerShell?

Simple PowerShell scripts without Windows specific components, such as these, to copy files, should work fine on Linux with `pwsh` installed and would work right out of the box with Windows.

guicaulada commented:
> Let's leave powershell out of this....

Sounds good. I'm not sure where the hate for PowerShell comes from, it's a native option on Windows and can be easily installed on non-Windows environments. That said, I'm fine leaving it out if we can come up with alternative solutions that reduce duplication, improve maintainability, and work across all environments.

Regarding the snippet you shared, I’m not sure I understand how it solves the problem. Would it work on both Windows and non-Windows systems?

Right now, there's a preference for solutions that work natively on Windows. But if we're open to compromise, we could even consider using Python, which is also easy to install across environments.
